### ISSUE TITLE
bug: cf worker shouldn't disconnect peers

### ISSUE THREAD

**Issue by ThaUnknown** [gfodor/p2pcf#15]
the CF worker really shouldn't disconnect peers, this should happen between the peers themselves

**Comment by gfodor** [comment]
hmm - the peer list is stored in R2 which has global strong consistency (read after write) except if there is a network partition. at least, that's what they say. so there shouldn't be any latency. If a peer that was once in the list is no longer there, then they timed out. if the timeout is too short, you can adjust it in the constructor.

**Comment by ThaUnknown** [comment]
i can assure you it's not consistent, anyways this should probably work like a BitTorrent tracker where it announces peers and doesn't dictate them

**Comment by gfodor** [comment]
can you describe exactly what is happening? because the removal case happens when a peer is in the list, and then is not in the list. it's not going to do that if there was latency in the peer arriving into the list.

**Comment by gfodor** [comment]
Are you seeing behavior inconsistent with what Cloudflare writes here?

![image](https://user-images.githubusercontent.com/220020/184558805-49489791-4842-4db7-bf95-c9a2ecc86317.png)

I'm not opposed to changing this, but I don't want to change it for the wrong reason.

**Comment by gfodor** [comment]
I think a better change is to keep the peer if it is connected. The scenario this is trying to cover is where you've joined a room and there are latent peers leftover from previous sessions that never disconnected gracefully, and you want to remove the unneeded RTCPeerConnections after they are timed out.

**Comment by ThaUnknown** [comment]
> can you describe exactly what is happening? because the removal case happens when a peer is in the list, and then is not in the list. it's not going to do that if there was latency in the peer arriving into the list.

the worker returns inconsistent peers from those who are expected to be in the lobby, this leads to people connecting to eachother then disconnecting the very next refresh, or even before they can finish connecting, the worker really shouldn't dictate which peers you should disconnect from

**Comment by gfodor** [comment]
yep - if you update this PR to just remove peers who are *not* currently connect I'll merge it. I want to leave this case in for the other one I mentioned: you join a room where there are still peers registered who didn't gracefully disconnect their sessions.

**Comment by gfodor** [comment]
oh now that you mention it I'm re-reading the code, and actually if there are a bunch of rapidly joining peers there might be some shuffling (despite R2's guarantees) since they do compete for slots in the list. the idea was that the state would eventually converge on subsequent polls. i never actually saw this happen in testing but i think what you're saying sounds possible with the current implementation, so i think I can cross of the concern that I don't understand what's going on.

in any case, update your PR and it will fix this problem.
